processors/data_contributions/processor.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import requests
+import logging
+from .. import base
+from . import extractors as extractors_module
+logger = logging.getLogger(__name__)
+
+
+def process(conf, conn):
+    def convert_ids(contrib):
+        converted_ids = {key: val.hex for key, val in contrib.items()
+                        if val.__class__.__name__ == 'UUID'}
+        contrib.update(converted_ids)
+        return contrib
+
+    extractors = base.helpers.get_variables(
+        extractors_module, lambda x: x.startswith('extract_'))
+
+    # Upsert source
+    source = extractors['extract_source']()
+    source_id = base.writers.write_source(conn, source)
+
+    # Upsert documents from data contributions
+    for_processing = """SELECT * FROM data_contributions
+                        WHERE approved=true
+                        AND trial_id IS NOT NULL
+                        AND document_category_id IS NOT NULL
+                        AND (data_url IS NOT NULL OR url IS NOT NULL);"""
+    for data_contrib in conn['explorer'].query(for_processing):
+        data_contrib = convert_ids(data_contrib)
+        trial = conn['database']['trials'].find_one(id=data_contrib['trial_id'])
+        if not trial:
+            continue
+        _process_document(conn, data_contrib, extractors, source_id)
+
+    # Remove documents from data_contributions
+    for_removal = """SELECT * FROM data_contributions
+                     WHERE approved=false
+                     AND document_id IS NOT NULL;"""
+    for data_contrib in conn['explorer'].query(for_removal):
+        data_contrib = convert_ids(data_contrib)
+        _remove_document(conn, data_contrib)
+
+
+def _process_document(conn, contrib, extractors, source_id):
+    document_id = None
+    document = extractors['extract_document'](contrib)
+    data_request = _request_data_contribution(document['source_url'])
+    if data_request:
+        content_type = data_request.headers['Content-Type']
+
+        # Check if the document is not an archive
+        if not any(ctype in content_type for ctype in ['-tar', '-gtar', 'zip', '7z', 'zlib']):
+            conn['database'].begin()
+            try:
+
+                # Upsert document
+                doc_category = conn['database']['document_categories'].find_one(
+                    id=document['document_category_id'])
+                document.update({
+                    'source_id': source_id,
+                    'name': doc_category['name'] if doc_category else None,
+                })
+                document_id = base.writers.write_document(conn, document)
+
+                # Update trial relation
+                conn['database']['trials_documents'].delete(document_id=document_id)
+                trial_rel = {
```

Comment:
Unless the word is very long, avoid abbreviations, as they can reduce readability. Reading the rest of the code I understand that `trial_rel` means `trial_relation`, but it's not instantly obvious when reading the name. Also, we already have a name for this: `trials_documents` (or, in singular, `trial_document`), so it's better to keep consistent naming.

As this variable is only used on the following `.insert()` call, you could even write it directly in the method call (i.e. `.insert({ ... })`).